x/auth/ante/unordered_test.go

```diff
@@ -5,140 +5,193 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
 
-	storetypes "cosmossdk.io/store/types"
-
+	"github.com/cosmos/cosmos-sdk/codec"
 	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
-	"github.com/cosmos/cosmos-sdk/runtime"
-	"github.com/cosmos/cosmos-sdk/testutil"
 	"github.com/cosmos/cosmos-sdk/testutil/testdata"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
 	"github.com/cosmos/cosmos-sdk/types/tx/signing"
 	"github.com/cosmos/cosmos-sdk/x/auth/ante"
-	authcodec "github.com/cosmos/cosmos-sdk/x/auth/codec"
-	"github.com/cosmos/cosmos-sdk/x/auth/keeper"
-	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	authsign "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+	txmodule "github.com/cosmos/cosmos-sdk/x/auth/tx/config"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 )
 
-func TestUnorderedAnte(t *testing.T) {
-	testPK, _, testAddr := testdata.KeyTestPubAddr()
+func TestSigVerification_UnorderedTxs(t *testing.T) {
+	var suite *AnteTestSuite
+	feeAmount := testdata.NewTestFeeAmount()
+	gasLimit := testdata.NewTestGasLimit()
+	var priv1, priv2, priv3 cryptotypes.PrivKey
+	var antehandler sdk.AnteHandler
+	var defaultSignMode signing.SignMode
+	var accs []sdk.AccountI
+	var msgs []sdk.Msg
+	reset := func(withUnordered bool) {
+		suite = SetupTestSuiteWithUnordered(t, true, withUnordered)
+		suite.txBankKeeper.EXPECT().DenomMetadata(gomock.Any(), gomock.Any()).Return(&banktypes.QueryDenomMetadataResponse{}, nil).AnyTimes()
+
+		enabledSignModes := []signing.SignMode{signing.SignMode_SIGN_MODE_DIRECT, signing.SignMode_SIGN_MODE_TEXTUAL, signing.SignMode_SIGN_MODE_LEGACY_AMINO_JSON}
+		// Since TEXTUAL is not enabled by default, we create a custom TxConfig
+		// here which includes it.
+		txConfigOpts := authtx.ConfigOptions{
+			TextualCoinMetadataQueryFn: txmodule.NewGRPCCoinMetadataQueryFn(suite.clientCtx),
+			EnabledSignModes:           enabledSignModes,
+		}
```

Comment:
do we need this?

Reply:
removed some stuff [slim down test copying](https://github.com/cosmos/cosmos-sdk/pull/24573/commits/14e628445641a309957cbe66d9c37184bb2cd3b3)